pkg/async/notifications/implementations/sendgrid_emailer.go

```diff
@@ -0,0 +1,67 @@
+package implementations
+
+import (
+	"context"
+	"os"
+
+	"github.com/sendgrid/sendgrid-go"
+	"github.com/sendgrid/sendgrid-go/helpers/mail"
+
+	"github.com/flyteorg/flyteadmin/pkg/async/notifications/interfaces"
+	runtimeInterfaces "github.com/flyteorg/flyteadmin/pkg/runtime/interfaces"
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/admin"
+	"github.com/flyteorg/flytestdlib/logger"
+	"github.com/flyteorg/flytestdlib/promutils"
+)
+
+type SendgridEmailer struct {
+	client        *sendgrid.Client
+	systemMetrics emailMetrics
+}
+
+func getEmailAddresses(addresses []string) []*mail.Email {
+	sendgridAddresses := make([]*mail.Email, len(addresses))
+	for idx, email := range addresses {
+		sendgridAddresses[idx] = mail.NewEmail("", email)
+	}
+	return sendgridAddresses
+}
+
+func getSendgridEmail(adminEmail admin.EmailMessage) *mail.SGMailV3 {
+	m := mail.NewV3Mail()
+
+	from := mail.NewEmail("Flyte Notifications", adminEmail.SenderEmail)
+	content := mail.NewContent("text/html", adminEmail.Body)
+	m.SetFrom(from)
+	m.AddContent(content)
+
+	personalization := mail.NewPersonalization()
+	emailAddresses := getEmailAddresses(adminEmail.RecipientsEmail)
+	personalization.AddTos(emailAddresses...)
+	personalization.Subject = adminEmail.SubjectLine
+	m.AddPersonalizations(personalization)
+
+	return m
+}
+
+func (s SendgridEmailer) SendEmail(ctx context.Context, email admin.EmailMessage) error {
+	m := getSendgridEmail(email)
+	s.systemMetrics.SendTotal.Inc()
+	response, err := s.client.Send(m)
+	if err != nil {
+		logger.Errorf(ctx, "Sendgrid error sending %s", err)
+		s.systemMetrics.SendError.Inc()
+		return err
+	}
+	s.systemMetrics.SendSuccess.Inc()
+	logger.Debugf(ctx, "Sendgrid sent email %s", response.Body)
+
+	return nil
+}
+
+func NewSendGridEmailer(config runtimeInterfaces.NotificationsConfig, scope promutils.Scope) interfaces.Emailer {
+	return &SendgridEmailer{
+		client:        sendgrid.NewSendClient(os.Getenv(config.NotificationsEmailerConfig.EmailerConfig.APIKeyEnvVar)),
```

Comment:
would it be worthwhile supporting different ways of passing the api key beyond just as an env var? for example, specifying a file path to the secret too? maybe something to worry about if/when someone requests so